video-app/src/app/store/videos/video.reducers.ts

```diff
@@ -0,0 +1,57 @@
+import * as VideoActions from './video.actions';
+import { VideoState, initializeState } from './video.state';
+import { Action, createReducer, on } from '@ngrx/store';
+import { Video } from '@core/models';
+
+export const initialState = initializeState();
+
+const reducer = createReducer(
+  initialState,
+  on(VideoActions.addYouTubeVideo, (state, { videoId }) => ({
+    ...state,
+    showErrorMessage: false,
+    videoId,
+  })),
+  on(VideoActions.addYouTubeVideoSuccess, (state, video: Video) => {
+    const videos = state.videoList.slice();

```

Comment:
ten slice aktualnie nic nie robi, może zamiast tego `[...state.videoList]`? od razu pozwoli to zgubić referencję:)